server/src/main/java/org/opensearch/indices/replication/checkpoint/ReplicationCheckpoint.java

```diff
@@ -30,37 +32,46 @@ public class ReplicationCheckpoint implements Writeable, Comparable<ReplicationC
     private final long segmentsGen;
     private final long segmentInfosVersion;
     private final long length;
+    private final String codec;
 
-    public static ReplicationCheckpoint empty(ShardId shardId) {
-        return new ReplicationCheckpoint(shardId);
+    public static ReplicationCheckpoint empty(ShardId shardId, CodecService codecService) {
+        return new ReplicationCheckpoint(shardId, codecService);
     }
 
-    private ReplicationCheckpoint(ShardId shardId) {
+    private ReplicationCheckpoint(ShardId shardId, CodecService codecService) {
         this.shardId = shardId;
         primaryTerm = SequenceNumbers.UNASSIGNED_PRIMARY_TERM;
         segmentsGen = SequenceNumbers.NO_OPS_PERFORMED;
         segmentInfosVersion = SequenceNumbers.NO_OPS_PERFORMED;
         length = 0L;
+        codec = codecService.codec("default").getName();
```

Comment:
Since all you need here is the name, I would recommend passing in a `String` instead of a `CodecService`. Let the caller figure out how to get the name from the CodecService. I _think_ this means you can probably avoid exposing the CodecService from IndexShard and instead only expose the codec service name.